RestaurantManagementSystem/RestaurantManagementSystem/src/UseCases/Register/RegisterRequestCommandHandler.cs

```diff
@@ -10,24 +10,21 @@ namespace RestaurantManagementSystem.src.UseCases.Register
 {
     public class RegisterRequestCommandHandler : IRegisterRequestCommandHandler
     {
-        private readonly IRequestRestaurant _requestRestaurant;
+        private readonly IRegistrationRequestRepo registrationRequestRepo;
         private readonly IEmailService _emailService;
 
-        public RegisterRequestCommandHandler(IRequestRestaurant requestRestaurant, IEmailService emailService)
+        public RegisterRequestCommandHandler(IRegistrationRequestRepo _registrationRequestRepo, IEmailService emailService)
         {
-            _requestRestaurant = requestRestaurant;
+            registrationRequestRepo = _registrationRequestRepo;
             _emailService = emailService;
         }
         public void Execute(string name, string email
-            , string address, ContactInfo contact)
+            , string address, string contactinfoemail, string contactinfophonenumber)
         {
-            //for specifing what random code contains
-            var options = new GenerationOptions(useNumbers: true, useSpecialCharacters: false, length: 8);
-            string verificationcode = ShortId.Generate(options);
-            RegistrationRequest obj = RegistrationRequest.Create(name,email,address,contact,RequestStatus.Status.PendingVerification,verificationcode);
-            _requestRestaurant.Add(obj);
-            string MessageBody = "You need to vertify your Email to Complete Your Request \n Your Request ID is : " + obj.Id + "\n" + " Your code is " + obj.VerificationCode;
-            _emailService.SendEmail(obj.Email, "Email Confirmation", MessageBody);
+            string verificationcode = registrationRequestRepo.GenerateVerificationCode();
+            RegistrationRequest obj = RegistrationRequest.Create(name,email,address,contactinfoemail, contactinfophonenumber, Status.PendingVerification,verificationcode);
```

Comment:
is this the layer that knows how the request should start? I mean the status? or this is a domain logic?